### DISCUSSION TITLE
Cast Shape

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#70] [Q&A] [answered]
I am constructing an `RShapeCast` for casting a shape. Does it take the position/rotation/direction values in world space?

I would like to cast a sphere from world Y:20 to world Y:-20. There is a floor (box) at zero level. My attempt doesn't detect a hit:
```js

const position = new Jolt.Vec3(0, 20, 0);
const rotation = new Jolt.Quat(0, 0, 0, 1);
const direction = new Jolt.Vec3(0, -40, 0);
const scale = new Jolt.Vec3(1, 1, 1);
const transform = new Jolt.Mat44();

transform.sTranslation(position);
transform.sRotation(rotation);
// not sure if scale is needed, but setting it as well
transform.sScale(scale);

const shapeCast = new Jolt.RShapeCast(shape, scale, transform, direction);

// using CastShapeClosestHitCollisionCollector for collector
system.GetNarrowPhaseQuery().CastShape(shapeCast, castSettings, offset, collector, bpFilter, objFilter, bodyFilter, shapeFilter);
```

Sime params for casting ray work fine, though. Ray can detect the floor and all the objects on top of it.

**Comment by jrouwe** [comment, marked as answer]
> transform.sTranslation(position);
> transform.sRotation(rotation);
> transform.sScale(scale);

Functions prefixed by an 's' are static functions, so they will not modify ```transform```, their purpose is to create a matrix.

You'll want to use:

```
const transform = Jolt.Mat44.prototype.sRotationTranslation(rotation, position);
```

or simply:

```
const transform = Jolt.Mat44.prototype.sTranslation(position);
```

since your rotation is identity.

This matrix cannot contain any scale (you're passing scale as a separate parameter to ```CastShape```).

**Reply by LeXXik** [reply]
Ah, those are statics, gotcha. Thank you! `sRotationTranslation` worked well:
https://playcanv.as/b/2fb1dc9d

I wanted to reuse Mat44, but looks like there is no way. I will destroy it immediately then.
